src/core_init_atmosphere/mpas_init_atm_static.F

```diff
@@ -306,66 +323,132 @@ subroutine init_atm_static(mesh, dims, configs)
+ ! point round off differences
+ !
+ ter_integer(:) = 0_I8KIND
+ ter(:) = 0.0_RKIND
+ nhs(:) = 0
 
-    end do
+ call mpas_pool_get_config(mgr % pool, 'tile_bdr', tile_bdr)
+ call mpas_pool_get_config(mgr % pool, 'tile_x', tile_nx)
+ call mpas_pool_get_config(mgr % pool, 'tile_y', tile_ny)
+
+ do i = 1, nCells
+     !
+     ! Load the tile at each cell center. This insures all cells recive values
+     !
+     if (nhs(i) == 0) then
+         tile => null()
+         ierr = mgr % get_tile(latCell(i), lonCell(i), tile)
+         if (ierr /= 0 .or. .not. associated(tile)) then
+             call mpas_log_write('Could not get tile that contained cell $i', intArgs=(/i/), messageType=MPAS_LOG_CRIT)
+         end if
+
+         ierr = mgr % push_tile(tile)
+         if (ierr /= 0) then
+             call mpas_log_write("Error pushing this tile onto the stack: "//trim(tile%fname), messageType=MPAS_LOG_CRIT)
+         end if
+     end if
+
+     !
+     ! Process each tile by removing it from the stack. Determine the closest cell center to each tile
+     ! pixel by using a KD search and assign its value to that cell.
+     !
+     do while (.not. mgr % is_stack_empty())
+         tile => mgr % pop_tile()
+
+         if (tile % is_processed) then
+             cycle
+         end if
+
+         call mpas_log_write('Processing tile: '//trim(tile%fname))
+
+         all_pixels_mapped_to_halo_cells = .true.
+
+         do j = 1 + tile_bdr, tile_ny + tile_bdr, 1
+            do k = 1 + tile_bdr, tile_nx + tile_bdr, 1
+
+               tval = tile % tile(k, j, 1)
+
+               call mgr % tile_to_latlon(tile, j, k, lat_pt, lon_pt)
+               call mpas_latlon_to_xyz(x, y, z, sphere_radius, lat_pt, lon_pt)
+               call mpas_kd_search(tree, (/x, y, z/), res)
+
+               !
+               ! For all but the outermost boundary cells, we can safely assume that the nearest
+               ! model grid cell contains the pixel (else, a different cell would be nearest)
+               !
+               if (bdyMaskCell(res % id) < nBdyLayers) then
+                  ter_integer(res % id) = ter_integer(res % id) + int(tval, kind=I8KIND)
+                  nhs(res % id) = nhs(res % id) + 1
+
+                  if (res % id <= nCellsSolve) then
+                      all_pixels_mapped_to_halo_cells = .false.
+                  end if
+
+               ! For outermost boundary cells, additional work is needed to verify that the pixel
+               ! actually lies within the nearest cell
+               else
+                  ! Model cell coordinates assume a "full" sphere radius
+                  ! at this point, so we need to use the same radius
+                  ! for source pixel coordinates
+                  call mpas_latlon_to_xyz(x, y, z, sphere_radius, lat_pt, lon_pt)
+
+                  if (mpas_in_cell(xPixel, yPixel, zPixel, xCell(res % id), yCell(res % id), zCell(res % id), &
```

Comment:
@mgduda The use of xPixel`, `yPixel`, and `zPixel` here was the cause of our problems. They need to be `x, y, z` respectively.